src/include/rados/rados_types.hpp

```diff
@@ -327,5 +327,27 @@ struct inconsistent_snapset_t {
  */
 const std::string all_nspaces(LIBRADOS_ALL_NSPACES);
 
+struct notify_ack_t {
+  uint64_t notifier_id;
+  uint64_t cookie;
+  std::string payload;
```

Comment:
Nit: RBD uses binary encoded payloads so a bufferlist or (if your trying to stick w/ the STL) a vector would be more appropriate.

Reply:
bufferlist sounds reasonable.